packages/yoastseo/src/parse/build/private/getSentencePositions.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Gets the start and end positions of all descendant nodes' tags and stores them in an array.
+ * Each object in the array represents an opening or closing tag.
+ * The startOffset and endOffset properties of the objects correspond to the start and end positions of the tags.
+ * Extracting this data into a separate array makes it easier to work with it (e.g. sort it and loop over it).
+ *
+ * @param {Node[]} descendantNodes	The descendant nodes to get tag positions from.
+ *
+ * @returns {Object[]}	An array of tag objects with structure { startOffset: [number], endOffset: [number] }.
+ *
+ */
+function getDescendantPositions( descendantNodes ) {
+	const descendantTagPositions = [];
+	descendantNodes.forEach( ( node ) => {
+		descendantTagPositions.push( node.sourceCodeLocation.startTag );
+		descendantTagPositions.push( node.sourceCodeLocation.endTag );
+	} );
+	// Sort the tag position objects by the start tag position in ascending order.
+	descendantTagPositions.sort( ( a, b ) => a.startOffset - b.startOffset );
+
+	return descendantTagPositions;
+}
+
+/**
+ *
+ * Adjusts the end position of the sentence to account for any descendant node tags that overlap with the sentence.
+ * We want to add the length of any tags that are within the sentence, as well as start tags at the beginning of the
+ * sentence and end tags at the end of the sentence, to the end position of the sentence.
+ * We don't want to include start tags at the end of the sentence in order to reduce the risk of invalid HTML after
+ * adding mark tags around the sentence. (Same goes for end tags at the start of the sentence, but this should already
+ * never happen, as these tags should be added to the end position of the previous sentence).
+ *
+ * @param {Node[]}		descendantNodes			The descendant nodes.
+ * @param {Object[]}	descendantTagPositions	The positions of the descendant nodes' tags.
+ * @param {number}		startPosition			The start position of a sentence.
+ * @param {number}		endPosition				The end position of a sentence.
+ *
+ * @returns {number}	The adjusted end position of the sentence.
+ */
+function adjustEndPosition( descendantNodes, descendantTagPositions, startPosition, endPosition ) {
+	/*
+	 * If the start position of a descendant's node tag is between the start and end position of the sentence, or is
+	 * the same as the start/end position of the sentence, add the tag's length to the end position of the sentence.
+	 */
+	descendantTagPositions.forEach( ( position ) => {
+		if ( position.startOffset >= startPosition && position.startOffset <= endPosition ) {
```

Comment:
(I'm assuming here that `startPosition` and `endPosition` will be renamed).

Reply:
@hansjovis I feel a bit on the fence about this because I don't feel like the lodash version is that much more readable or concise. And I thought it was good to rely on external dependencies (including lodash) as little as possible? What is in your opinion an advantage of using `inRange`?

Reply:
Hmmm... Now I'm on the fence as well.

I agree with your assessment that the lodash version is not that much more readable. Maybe even less so, since it is a bit confusing which argument is the value to check, and which is the minimum boundary and maximum boundary.

I would say we keep it as is, and not use `inRange` 🙂 .